spec/forms/state_file/nj_estimated_tax_payments_form_spec.rb

```diff
@@ -6,28 +6,91 @@
   describe "validations" do
     let(:form) { described_class.new(intake, params) }
 
-    it_behaves_like :nj_money_field_concern, field: :estimated_tax_payments, can_be_empty: true do
+    it_behaves_like :nj_money_field_concern, field: :estimated_tax_payments, can_be_empty: false do
       let(:form_params) do
-        { estimated_tax_payments: money_field_value }
+        {
+          has_estimated_payments: "yes",
+          overpayments: 0,
+          estimated_tax_payments: money_field_value
+        }
+      end
+    end
+
+    it_behaves_like :nj_money_field_concern, field: :overpayments, can_be_empty: false do
+      let(:form_params) do
+        {
+          has_estimated_payments: "yes",
+          overpayments: money_field_value,
+          estimated_tax_payments: 0
+        }
+      end
+    end
+
+    context "invalid params" do
+      context "has_estimated_payments is required" do
+        let(:params) do
+          { has_estimated_payments: nil }
+        end
+
+        it "is invalid" do
+          expect(form.valid?).to eq false
+          expect(form.errors[:has_estimated_payments]).to include "Can't be blank."
+        end
+      end
+
+      context "estimated_tax_payments is required if has_estimated_payments=yes" do
+        let(:params) do
+          {
+            has_estimated_payments: "yes",
+            overpayments: 0,
+            estimated_tax_payments: nil
+          }
+        end
+
+        it "is invalid" do
+          expect(form.valid?).to eq false
+          expect(form.errors[:estimated_tax_payments]).to include "Can't be blank."
+        end
+      end
+
+      context "overpayments is required if has_estimated_payments=yes" do
+        let(:params) do
+          {
+            has_estimated_payments: "yes",
+            overpayments: nil,
+            estimated_tax_payments: 0
+          }
+        end
+
+        it "is invalid" do
+          expect(form.valid?).to eq false
+          expect(form.errors[:overpayments]).to include "Can't be blank."
+        end
```

Comment:
[pebble] pretty sure these are redundant with the shared example `nj_money_field_concern`